app/controllers/videos_controller.rb

```diff
@@ -0,0 +1,33 @@
+class VideosController < ApplicationController
+  decorates_assigned :video, with: ContentItemDecorator
+  decorates_assigned :related_content, with: CategoryDecorator
+
+  include Navigation
+
+  def show
+    @video = interactor.call do
+      not_found
+    end
+
+    set_related_content
+    set_categories
+  end
+
+  private
+
+  def interactor
+    Core::VideoReader.new(params[:id])
+  end
+
+  def set_related_content
+    @related_content = CategoriesWithRestrictedContents.build(@video.categories,
```

Comment:
Stylistically it might be better to break each argument to `build` on a new line. i.e.

``` ruby
BlehThing.build(
  @video.categories,
  @something.things
)    
```